### PR TITLE
Update tastypie

### LINKED ISSUE #3570: Tastypie needs to upgrade our dependencies to support Python3
If you go to http://localhost:8000/api/v1/project/46162/sync_versions/ while running your instance with Python3 you will get this error:

> 'dict' object has no attribute 'has_key'

I found that we are using `mimetype`, https://github.com/rtfd/readthedocs.org/blob/df8079a18ed9c15e9c85f0646e3e9169e725b310/requirements/pip.txt#L49 and `python-mimeparse` should be used.

We should write a simple test for this, otherwise we will find this out in production :)

Ref: http://django-tastypie.readthedocs.io/en/latest/python3.html#changed-requirements

### PR DESCRIPTION
**Pull request by stsewd**
Ref to https://github.com/rtfd/readthedocs.org/pull/4319#issuecomment-402039200 and https://github.com/rtfd/readthedocs.org/issues/3570#issuecomment-401184862

Closes #3570

### CODE REVIEW

File: readthedocs/api/base.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 from django.core.cache import cache
 from django.shortcuts import get_object_or_404
 from tastypie import fields
-from tastypie.authorization import DjangoAuthorization
+from tastypie.authorization import DjangoAuthorization, ReadOnlyAuthorization
```
Comment (stsewd) [inline readthedocs/api/base.py]: There are other resources using `DjangoAuthorization`, but tests were failing only for the `ProjectResource`

File: readthedocs/api/base.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ class Meta(object):
         allowed_methods = ['get', 'post', 'put']
         queryset = Project.objects.api()
         authentication = PostAuthentication()
-        authorization = DjangoAuthorization()
+        authorization = ReadOnlyAuthorization()
```
Comment (stsewd) [inline readthedocs/api/base.py]: Authorization is already default to `ReadOnlyAuthorization`, I'm just being explicit here.

File: readthedocs/rtd_tests/tests/test_api.py [hunk 3]
```diff
@@ -417,8 +417,8 @@ def test_get_invalid_raw_log(self):
 class APITests(TestCase):
     fixtures = ['eric.json', 'test_data.json']
 
-    def test_make_project(self):
-        """Test that a superuser can use the API."""
+    def test_cant_make_project(self):
+        """Test that a can't use the API to create projects."""
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_api.py]: We aren't using this in our code, I'm not sure if we allow this in the external API, we don't have that documented either https://docs.readthedocs.io/en/latest/api/v1.html

File: readthedocs/api/base.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/api/base.py]: I think we don't have tests for the other resources, that's why

**Comment by stsewd** [comment]
Ok, this is weird, tests are passing for 2d59d29 on travis, but locally (I just checked again!)  are falling. Anyone else can check that? (yes, I'm running my tests with `tox -r`)

**Comment by agjohnson** [comment]
This looks tied to a version `3.0` release, along with django 1.10. Targeting our `3.0` release for this, I think we can de-prioritize this work until we've freed up some more resources.

**Comment by stsewd** [comment]
Last time I check https://github.com/rtfd/readthedocs.org/pull/4325#issuecomment-402285957 tests were passing locally btw

File: requirements/pip.txt [hunk 4]
```diff
@@ -15,7 +15,7 @@ six==1.11.0
 future==0.16.0
 
 # django-tastypie 0.13.x and 0.14.0 are not compatible with our code
```
Comment (davidfischer) [inline requirements/pip.txt]: Let's update this comment. From the [tastypie changelog](https://django-tastypie.readthedocs.io/en/latest/release_notes/v0.13.3.html), it looks like 0.13.3 is still compatible with Django 1.9.

**Comment by stsewd** [comment]
I have updated tastypie to 0.13.3 and rebased, tests should pass (hope p:)

**Comment by davidfischer** [comment]
I ran a manual test against the endpoints in the v1 API docs and they looked ok to me. The only exception was the [File Search API](https://docs.readthedocs.io/en/latest/api/v1.html#file-search) which doesn't actually appear to be implemented correctly. This wasn't an issue with this PR though. It isn't working in production or with the old tastypie.

**Comment by stsewd** [comment]
@humitos this PR is already included in https://github.com/rtfd/readthedocs.org/pull/4750

**Comment by davidfischer** [comment]
This is included in #4750 but I believe it could be merged anytime. The Django 1.11 upgrade relies on this but this can be merged and used with our current setup.

**Pull request merged**